### PR TITLE
Remove omitempty in order to permit update with empty values

### PR DESCRIPTION
**Pull request by Galiley**
The API already handle empty and null values

### CODE REVIEW

**Comment by Serpentiel** [comment]
hey, @Galiley! 👋

thank you for your contribution!

could you please elaborate a bit further, why would you want to send empty/nil values to the API anyway?

**Comment by Galiley** [comment]
I want to send empty/null values to the API in order to remove for instance all the channels ACL for a certain user. This option is available in the Aiven Console

**Comment by Serpentiel** [comment]
thanks! I will have to evaluate whether there are potential problems which could be caused by this change, particularly in other Aiven products that are dependent on this library, I will get back to you soon

**Comment by Serpentiel** [comment]
afaik `omitempty` was supposed to introduce a similar behavior

**Comment by Galiley** [comment]
Quoted from the documentation of [json.Marshal](https://pkg.go.dev/encoding/json#Marshal)
> The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string. 

Empty slice, array or map is not encoded with omitempty which are supported by the Aiven API

File: service_user.go [hunk 1]
```diff
@@ -59,6 +61,27 @@ type (
 	}
 )
 
+func (ac AccessControl) MarshalJSON() ([]byte, error) {
```
Comment (Serpentiel) [inline service_user.go:64]: please comment what this function does

File: service_user_test.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package aiven
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAccessControlMarshalJSON(t *testing.T) {
```
Comment (Serpentiel) [inline service_user_test.go:10]: please make the use of `testing.T` as we do in the other tests

**Review by Serpentiel — CHANGES_REQUESTED** [review]
this PR introduces no misbehavior for the dependent projects, although there are some changes that would be neat to have

**Review by Serpentiel — APPROVED** [review]
lgtm

**Pull request merged**